tekton/ci/jobs/conditions.yaml

```diff
@@ -62,3 +62,68 @@ spec:
       REGEX="$(echo $(params.gitHubCommand) | awk '{ print $2}')"
       [[ "$REGEX" == "" ]] && REGEX='.*'
       echo "$(params.checkName)" | grep -E "$REGEX"
+---
+# These tasks replace the above conditions with Tasks that have results which can be used in when expressions
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: check-git-files-changed
+  namespace: tektonci
+  annotations:
+    description: |
+      Returns the number of files (matching the regex) that have changed in the provided git repo
+spec:
+  params:
+    - name: gitCloneDepth
+      description: Number of commits + 1
+    - name: regex
+      description: Regular expression to match files changed
+  workspaces:
+    - name: source
+  results:
+    - name: numFilesChanged
+      description: |
+        Contains the number of files that have changed in the last N commits from
+        the revision of the git repo that is available in the source workspace
+  steps:
+    - name: count-files
+      image: alpine/git
+      script: |
+        set -o pipefail
+
+        BACK="HEAD~$(( $(params.gitCloneDepth) - 1 ))"
+
+        cd $(workspaces.source.path)
```

Comment:
No tests make me sad, but I'm not sure if there's a better way to do this for something so simple. :(

Reply:
now that we have both params + results (and workspaces i guess XD) it be cool if we could provide some tools that did some simple input output tests 🤔  (ala https://github.com/tektoncd/pipeline/issues/1289) - would be harder for tasks and pipelines with side effects but we could start with the ones that dont and work our way through it (maybe use hermetic mode to suppress side effects? and or use something like workflows to make it easier to isolate the parts with side effects...)

(as a side note when i started working on updating the triggering bits, i realized that a lot of the problems i ran into could have been easily caught by some kind of static analysis - e.g. i initially removed the embedded pipeline resource but forgot to add the required workspace - would have been super cool if my IDE could have just been like "hey you are missing a required workspace here...")